### PR TITLE
task builder replace taskInfo construction logic

### PR DESCRIPTION
**Pull request by cmingxu**
Signed-off-by: kevin xu <cming.xu@gmail.com>

### CODE REVIEW

**Comment by xychu** [comment]
+1

File: src/manager/framework/state/task_builder.go [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+	builder.taskInfo = &taskInfo
+
+	return builder
+}
+
+func (builder *TaskBuilder) GetTaskInfo() *mesos.TaskInfo {
+	return builder.taskInfo
+}
+
+func (builder *TaskBuilder) SetName(name string) *TaskBuilder {
```
Comment (upccup) [inline src/manager/framework/state/task_builder.go]: is this functions need return value?

File: src/manager/framework/state/task_builder.go [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+
+func (builder *TaskBuilder) SetLabels(labelMap map[string]string) *TaskBuilder {
+	labels := make([]*mesos.Label, 0)
+	for k, v := range labelMap {
+		labels = append(labels, &mesos.Label{
+			Key:   proto.String(k),
+			Value: proto.String(v),
+		})
+
+		builder.taskInfo.Labels = &mesos.Labels{
```
Comment (upccup) [inline src/manager/framework/state/task_builder.go]: maybe put this out of for loop

File: src/manager/framework/state/task_builder.go [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+				}
+			}
+		}
+
+		protocol := strings.ToLower(healthCheck.Protocol)
+		if protocol == "http" {
+			builder.taskInfo.HealthCheck = &mesos.HealthCheck{
+				Type: mesos.HealthCheck_HTTP.Enum(),
+				Http: &mesos.HealthCheck_HTTPCheckInfo{
+					Scheme:   proto.String("http"),
```
Comment (upccup) [inline src/manager/framework/state/task_builder.go]: Scheme:   proto.String("http"), ==> Scheme:   proto.String(protocol)

**Comment by upccup** [comment]
@cmingxu if this implement named builder. i think rename setxxx to buildxxx is better.

**Pull request merged**